private-templates-service/client/src/components/Dashboard/Dashboard.tsx

```diff
@@ -51,29 +61,35 @@ class Dashboard extends React.Component<Props, State> {
     props.getTemplates();
   }
 
+  selectTemplate = (templateID: string) => {
+    this.props.getTemplate(templateID);
+    this.toggleModal();
+  }
+
   toggleModal = () => {
     this.setState({ isPreviewOpen: !this.state.isPreviewOpen });
   };
 
   render() {
-    if(this.props.isSearch) {
-      return(
+    if (this.props.isSearch) {
+      return (
         <DashboardContainer>
-          <SearchPage/>
+          <SearchPage />
         </DashboardContainer>
       );
     }
     //TODO add sort functionality to separate templates displayed in recent vs draft
-    let templates = undefined;
-    if (!this.props.batchTemplates.isFetching && this.props.batchTemplates.templateList) {
-      templates = this.props.batchTemplates.templateList.templates;
+    let templates = new Array<Template>();
+    if (!this.props.templates.isFetching
+      && this.props.templates.templates
```

Comment:
This part is a little fuzzy for me. Why you have multiple levels of the same type? or is it a naming issue?

Reply:
yeah, really unfortunate naming here.

`this.props.templates` is what I named the variable.
`this.props.templates.templates` is a `TemplateList` object, containing a list as well as some additional data
`this.props.templates.templates.templates` is a 'Templates[]` object.